src/snowflake/snowpark/functions.py

```diff
@@ -829,6 +964,9 @@ def create_map(*cols: Union[ColumnOrName, Iterable[ColumnOrName]]) -> Column:
         -----------------------
         <BLANKLINE>
     """
+
```

Comment:
The type hint is correct although it could be tough to understand. It allows the following

```
create_map(col_key, col_value)
create_map([col_key, col_value])
create_map({col_key, col_value})
create_map(col_key1, col_value1, col_key2, col_value2)
```

Reply:
I think I am now prob. left with more questions. I'd actually suggest in our meeting to clarify what the correct type hints should be and then introduce a helper function
`_check_map_parameter(cols)` to validate it. This could be helpful.

Reply:
for typehint:

> Union[ColumnOrName, List[ColumnOrName], Tuple[ColumnOrName]]